pkg/scheduler/plugins/gang/gang_cache.go

```diff
@@ -0,0 +1,383 @@
+package gang
+
+import (
+	"github.com/koordinator-sh/koordinator/apis/extension"
+	"github.com/koordinator-sh/koordinator/pkg/util"
+	"k8s.io/api/core/v1"
+	"k8s.io/klog/v2"
+	"strconv"
+	"sync"
+	"time"
+)
+
+type gangCache struct {
+	lock      *sync.RWMutex
+	gangItems map[string]*Gang
+}
+
+func NewGangCache() *gangCache {
+	return &gangCache{
+		gangItems: make(map[string]*Gang),
+		lock:      new(sync.RWMutex),
+	}
+}
+
+func (gangCache *gangCache) onPodAdd(obj interface{}) {
+	pod, ok := obj.(*v1.Pod)
+	if !ok {
+		return
+	}
+	gangCache.AddPod(pod)
```

Comment:
function should be: first get gang, then execute gang's function. Don't put all logic to gangCache, gangCache only maintain gang's add\update\delete

Reply:
done